packages/devtools_app/lib/src/flutter/controllers.dart

```diff
@@ -144,10 +145,23 @@ class _InheritedProvider extends InheritedWidget {
       oldWidget.data != data;
 }
 
-/// Superclass for all DevTools controller providers.
+/// Controller provider that [DevToolsScreen] instances can provide.
 ///
-/// For an example implementation, see `src/example/provided_controller.dart`.
-abstract class ControllerProvider extends StatefulWidget {
-  const ControllerProvider({Key key, this.child}) : super(key: key);
+/// Flutter UI components with this [ControllerProvider] as an ancestor can
+/// access the provided controller via `Provider.of<T>(context)`.
+class ControllerProvider<T> extends StatelessWidget {
```

Comment:
We can remove this ControllerProvider class as it isn't providing value beyond the base Provider class. Instead we should just pass in the `create` Functions which gives a cleaner API anyway.